text/0000-type-inference.md

```diff
@@ -0,0 +1,168 @@
+# FusionJS Plugin Type Inference
```

Comment:
This is great!  Thanks @ganemone for putting this together.  My personal opinion here is that the trade-off for a better developer experience is worth the loss of type coverage.

It is unfortunate that Flow in its present state lacks global inference which would solve this problem without having to resort to workarounds like this.

Reply:
To be fair to flow, it seems they have some good reasons for not supporting global type inference which will help performance in a monorepo, but it would be nice.